catalog/harvest/catalog-harvest-metacardadapter/src/main/java/org/codice/ddf/catalog/harvest/adapter/MetacardOnlyAdapter.java

```diff
@@ -0,0 +1,202 @@
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.harvest.adapter;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+import ddf.security.service.SecurityServiceException;
+import java.io.Serializable;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.Validate;
+import org.codice.ddf.catalog.harvest.HarvestException;
+import org.codice.ddf.catalog.harvest.HarvestedResource;
+import org.codice.ddf.catalog.harvest.StorageAdapter;
+import org.codice.ddf.security.common.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Harvester {@link StorageAdapter} for creating {@link Metacard}s in the catalog. */
+public class MetacardOnlyAdapter implements StorageAdapter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MetacardOnlyAdapter.class);
+
+  private static final Security SECURITY = Security.getInstance();
+
+  private final CatalogFramework catalogFramework;
+
+  private final HarvestedResourceTransformer harvestedResourceTransformer;
+
+  public MetacardOnlyAdapter(
+      CatalogFramework catalogFramework,
+      HarvestedResourceTransformer harvestedResourceTransformer) {
+    Validate.notNull(catalogFramework, "argument {catalogFramework} cannot be null");
+    Validate.notNull(
+        harvestedResourceTransformer, "argument {harvestedResourceTransformer} cannot be null");
+
+    this.catalogFramework = catalogFramework;
+    this.harvestedResourceTransformer = harvestedResourceTransformer;
+  }
+
+  @Override
+  public String create(HarvestedResource resource) throws HarvestException {
+    Metacard metacard = harvestedResourceTransformer.transformHarvestedResource(resource);
+
+    if (metacard == null) {
```

Comment:
❓ Should we throw the exception straight from `harvestedResourceTransformer.transformHarvestedResource(resource)` instead of checking for `null` and throwing here?